### PR TITLE
indra, indra-publishing: GPL-3.0-only license option

### PR DESCRIPTION
**Pull request by kashike**
fixes GH-17

### CODE REVIEW

File: indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt [hunk 1]
```diff
@@ -117,6 +117,13 @@ open class IndraExtension @Inject constructor(objects: ObjectFactory) {
     url = "https://opensource.org/licenses/Apache-2.0"
   ))
 
+  fun gpl30OnlyLicense() = this.license.set(License(
+    spdx = "GPL-3.0-only",
+    name = "GNU General Public License version 3",
+    url = "https://opensource.org/licenses/GPL-3.0",
+    bintray = "GPL-3.0"
```
Comment (kashike) [inline indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt:124]: Bintray is silly and does not use SPDX identifiers: https://www.jfrog.com/confluence/display/BT/Bintray+REST+API#BintrayRESTAPI-DeleteDockerTag

File: indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt [hunk 2]
```diff
@@ -117,6 +117,13 @@ open class IndraExtension @Inject constructor(objects: ObjectFactory) {
     url = "https://opensource.org/licenses/Apache-2.0"
   ))
 
+  fun gpl30OnlyLicense() = this.license.set(License(
+    spdx = "GPL-3.0-only",
+    name = "GNU General Public License version 3",
```
Comment (kashike) [inline indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt:122]: SPDX adds " only" to the end of the name, but I don't think we should?

File: indra-common/src/main/kotlin/net/kyori/indra/data/License.kt [hunk 3]
```diff
@@ -26,5 +26,6 @@ package net.kyori.indra.data
 data class License(
   val spdx: String?,
   val name: String,
-  val url: String
+  val url: String,
+  val bintray: String? = spdx
```
Comment (kashike) [inline indra-common/src/main/kotlin/net/kyori/indra/data/License.kt:30]: @zml2008 does this work as I expect? or do we need to do something to allow groovy to not die?

Comment (zml2008) [reply indra-common/src/main/kotlin/net/kyori/indra/data/License.kt:30]: you'll need to add a `@JvmOverloads constructor` before the parameters

File: indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt (hunk 2 above)
Comment (zml2008) [reply indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt:122]: yeah, I agree

File: indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt [hunk 4]
```diff
@@ -117,6 +117,13 @@ open class IndraExtension @Inject constructor(objects: ObjectFactory) {
     url = "https://opensource.org/licenses/Apache-2.0"
   ))
 
+  fun gpl30OnlyLicense() = this.license.set(License(
```
Comment (zml2008) [inline indra-common/src/main/kotlin/net/kyori/indra/IndraExtension.kt:120]: not a fan of the `30` part. should indicate the decimal somehow, or skip the `.0` -- the FSF seem to use `v3` alone in most places, with the `.0` only in their URL.

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Review by zml2008 — APPROVED** [review]

**Pull request merged**